lib/concurrent-ruby/concurrent/executor/safe_task_executor.rb

```diff
@@ -16,20 +16,20 @@ def initialize(task, opts = {})
 
     # @return [Array]
     def execute(*args)
-      synchronize do
-        success = false
-        value   = reason = nil
+      success = true
+      value   = reason = nil
 
+      synchronize do
         begin
           value   = @task.call(*args)
           success = true
         rescue @exception_class => ex
           reason  = ex
           success = false
         end
-
-        [success, value, reason]
       end
+
+      [success, value, reason]
```

Comment:
I don't understand this change, it should never make a difference as far as I can tell.
All possibilities are:
* @task.call does not raise an exception, so `success = true`
* @task.call raises something caught by `rescue @exception_class`, so `success = false`
* @task.call raises something NOT caught by `rescue @exception_class` (e.g., like a break or return or exception higher in the hirearchy), then there should be no way we would ever reach the end of this method, or even the end of the synchronize block.

So the only case I could see this would make a difference is a bug in CRuby where it break or return to the semantically incorrect place, such as after that synchronize call (which should only be possible with a `break` syntactically inside the synchronize block, and there isn't any).

That said, this change is not hurting either, so I think it's fine to keep the change, but we should have a better understanding why it's needed, and a clearer test (see https://github.com/ruby-concurrency/concurrent-ruby/issues/931#issuecomment-1069094690).